controlplane/eks/api/v1beta1/types.go

```diff
@@ -250,6 +252,8 @@ type OIDCIdentityProviderConfig struct {
 	// and must be publicly accessible over the internet.
 	//
 	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:Pattern=`^https://.+`
```

Comment:
Might be better to use a CEL validation if possible, `isURL` likely more comprehensive

Reply:
Exactly, and then you can add `,message="..."` with a custom error message too